src/checker.ml

```diff
@@ -834,7 +834,7 @@ let[@inline] touch_with_index (state: checker) (index:Ligo.tez) : (LigoOp.operat
 let entrypoint_touch (state, _: checker * unit) : (LigoOp.operation list * checker) =
   let index = match state.last_price with
     | None -> state.parameters.index (* use the old one *)
-    | Some i -> Ligo.mul_nat_tez i (Ligo.tez_from_literal "1mutez") in (* FIXME: Is the nat supposed to represent tez? *)
+    | Some i -> i in (* FIXME: Is the nat supposed to represent tez? *)
```

Comment:
Re: the FIXME. It seems that `state.last_price` is meant to have units tez/kit, right?

Reply:
Well, I am still not entirely sure, that's why I've been reluctant to remove the `FIXME` here :thinking: I'll make a pass and see if I can (re-)figure out all the units of measure again in a separate PR, given how they are used.